File: Android_Sprint2/app/src/main/java/com/example/israel/android_sprint2/MainActivity.java
package com.example.israel.android_sprint2;

import android.graphics.drawable.Drawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ArrayList<ShoppingItem> shoppingItems;
    private ShoppingListAdapter shoppingListAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // initialize shared prefs DAO
        SelectedShoppingItemsSPDAO.init(this);

        // initialize shopping items
        shoppingItems = new ArrayList<>(ShoppingItemConstants.ICON_IDS.length);
        for (int i = 0; i < ShoppingItemConstants.ICON_IDS.length; ++i) {
            String name = ShoppingItemConstants.ITEM_NAMES_RAW[i];
            ShoppingItem newShoppingItem = new ShoppingItem(ShoppingItemConstants.ICON_IDS[i], name, Integer.toString(i));
            shoppingItems.add(newShoppingItem);
        }

        // TODO: IRRELEVANT: sort shoppingItems

        // TODO: select saved selected shopping items

        // set up recycler view
        RecyclerView recyclerView = findViewById(R.id.recycler_view_shopping_list);
        recyclerView.setHasFixedSize(true);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        recyclerView.setLayoutManager(layoutManager);

        shoppingListAdapter = new ShoppingListAdapter(this, shoppingItems);
        recyclerView.setAdapter(shoppingListAdapter);

    }
}

File: app/src/main/java/com/lambdaschool/sprint2_challenge/ui/MainActivity.kt
package com.lambdaschool.sprint2_challenge.ui
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.lambdaschool.sprint2_challenge.R
import com.lambdaschool.sprint2_challenge.utils.ShoppingItemConstants
import com.lambdaschool.sprint2_challenge.model.ShoppingItem
import java.util.*

class MainActivity : AppCompatActivity() {

    internal var shoppingList: ArrayList<ShoppingItem> = ArrayList()
    private val adapter = ShoppingListAdapter(shoppingList)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
       val shoppingListLayout=findViewById<RecyclerView>(R.id.shopping_list_layout)
        shoppingListLayout.setHasFixedSize(false)
        val manager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        shoppingListLayout.layoutManager = manager
        shoppingListLayout.adapter = adapter

        for (i in ShoppingItemConstants.ICON_IDS) {
            shoppingList.add(ShoppingItem(ShoppingItemConstants.ITEM_NAMES_RAW[i], ShoppingItemConstants.ICON_IDS[i], 1))

        }
        adapter.notifyDataSetChanged()
    }
}
